controllers/etcd_custodian_controller.go

```diff
@@ -220,10 +220,10 @@ func calculatePDBminAvailable(etcd *druidv1alpha1.Etcd) int {
 			}
 		}
 		values = append(values, readyMembers)
-	}
-
-	if backupReady {
-		values = append(values, clusterSize)
+	} else {
```

Comment:
The two conditions `!backReady` and `!allMembersReady` are disjoint and there is no reason to use it in if..else block.
Just have:
```go
if !backupReady {
	values = append(values, clusterSize)
}
if !allMembersReady {
	readyMembers := 0
	for _, member := range etcd.Status.Members {
		if member.Status == druidv1alpha1.EtcdMemberStatusReady {
			readyMembers += 1
		}
	}
	values = append(values, readyMembers)
}
```
In the code next you anyways take the max out of all the entries in the `value` slice, right?

Reply:
If backReady = false and allMembersReady = false then in the proposed change you will only enter the `if` block and not the `else` block. If the backup is not ready then by setting `minAvailable` to `clusterSize` you essentially wish to prevent update of any member in the cluster. This would not happen then.

Reply:
I don't consider both conditions being disjoint in that case: It doesn't make sense to use the entire cluster size as `minAvailable` if both not all members and backups aren't ready, i.e. we should rather use the lower number in such a case to allow disruptions in case we want to ship a fix.

I even thought about disregarding the backup condition and remove it from the calculation because there is no benefit but rather disadvantages: If backups aren't ready because of a software issue, then the PDB would forbid rolling out a fix to the cluster. WDYT?

Reply:
>we should rather use the lower number in such a case to allow disruptions in case we want to ship a fix.

If that is the case then there could be a bug due to which `backReady` will be false. Assume all members are ready and backReady is false in that case you will still set the minAvailable to cluster size thus preventing any fix to be applied which might correct the issue with taking backups. Isn't it?

Reply:
I am fine with leaving out backupReady condition check altogether. Also we should force a re-election via the backup-restore sidecar container.

Reply:
> Today that is not done. Setting the minAvailable to clusterSize in that case will result in a situation where no backup is taken and you cannot even update the etcd cluster with a fix.

That's exactly the reason I think considering the backup condition isn't beneficial.

Reply:
I think it is just a safety net but without re-election this safety net is quite useless. The only case i can think of is if there is only a transient error in taking backups then retries from the leading backup-restore sidecar will happen. While this is happening no updates should be allowed to give the leading sidecar a slim chance to salvage data before updates roll in. Removing this would also remove this safety net :)